packages/typescript/ai/tests/middlewares/memory.test.ts

```
@@ -0,0 +1,567 @@
+import { memoryMiddleware } from '../../src/memory'
+import type {
+  MemoryAdapter,
+  MemoryHit,
+  MemoryListResult,
+  MemoryQuery,
+  MemoryRecord,
+  MemoryScope,
+  MemorySearchResult,
+} from '../../src/memory'
+import type { StreamChunk } from '../../src/types'
+import { ev, createMockAdapter, collectChunks } from '../test-utils'
```

ESLint

[error] `../test-utils` import should occur before type import of `../../src/memory`

(import/order)